tests/e2e/scenarios/node-remove.go

```diff
@@ -0,0 +1,212 @@
+/*
+Copyright © 2020 Dell Inc. or its subsidiaries. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package scenarios
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/onsi/ginkgo"
+	corev1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2elog "k8s.io/kubernetes/test/e2e/framework/log"
+	storageframework "k8s.io/kubernetes/test/e2e/storage/framework"
+
+	"github.com/dell/csi-baremetal-e2e-tests/e2e/common"
+)
+
+// DefineNodeRemovalTestSuite defines custom csi-baremetal node removal test
+func DefineNodeRemovalTestSuite(driver *baremetalDriver) {
+	ginkgo.Context("Baremetal-csi node remove test", func() {
+		defineNodeRemovalTest(driver)
+	})
+}
+
+func defineNodeRemovalTest(driver *baremetalDriver) {
+	ginkgo.BeforeEach(skipIfNotAllTests)
+
+	var (
+		pod           *corev1.Pod
+		pvc           *corev1.PersistentVolumeClaim
+		k8sSC         *storagev1.StorageClass
+		executor      = common.GetExecutor()
+		driverCleanup func()
+		ns            string
+		taintNodeName string
+		ctx           context.Context
+		f             = framework.NewDefaultFramework("node-remove-test")
+	)
+
+	init := func() {
+		var (
+			perTestConf *storageframework.PerTestConfig
+			err         error
+		)
+		ns = f.Namespace.Name
+		ctx = context.Background()
+		perTestConf, driverCleanup = driver.PrepareTest(f)
+		k8sSC = driver.GetDynamicProvisionStorageClass(perTestConf, "xfs")
+		k8sSC, err = f.ClientSet.StorageV1().StorageClasses().Create(context.TODO(), k8sSC, metav1.CreateOptions{})
+		framework.ExpectNoError(err)
+	}
+
+	cleanup := func() {
+		e2elog.Logf("Starting cleanup for test NodeRemoval")
+
+		if taintNodeName != "" {
+			_, _, err := executor.RunCmd(fmt.Sprintf("docker start %s", taintNodeName))
```

Comment:
Could we check that csi-baremetal-node pod was successfully created and ready on the restored node?